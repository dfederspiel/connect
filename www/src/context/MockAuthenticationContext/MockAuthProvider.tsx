import { useProvideMockAuth } from '.';
import { AuthProvider } from '../AuthenticationContext/';
import { AuthContext } from '../AuthenticationContext/types';
// import { AuthModule, Mode } from "./AuthModule";
/**
 * AuthContext to be passed to any component that depends on user context.
 */
// const authContext = createContext<AuthContext>(null as any);

// Provider component that wraps your app and makes auth object ...
// ... available to any child component that calls useAuth().
// eslint-disable-next-line react/prop-types

interface MockAuthProviderProps {
  children?: JSX.Element;
  user?: string;
}

export const MockAuthProvider = ({
  children,
  user,
}: MockAuthProviderProps): JSX.Element => {
  const auth = useProvideMockAuth(user) as AuthContext;
  return <AuthProvider provider={auth}>{children}</AuthProvider>;
};
