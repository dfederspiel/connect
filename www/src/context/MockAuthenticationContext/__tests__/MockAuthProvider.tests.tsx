import { render } from '@testing-library/react';
import { MockAuthProvider } from '../MockAuthProvider';

describe('the mocked auth provider', () => {
  it('exists', () => {
    const { asFragment } = render(<MockAuthProvider user="Joe Black"></MockAuthProvider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
