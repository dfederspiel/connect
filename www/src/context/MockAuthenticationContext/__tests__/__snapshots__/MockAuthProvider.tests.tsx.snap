// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the mocked auth provider exists 1`] = `<DocumentFragment />`;
